gammapy/estimators/map/ts.py

```diff
@@ -479,36 +483,56 @@ def run(self, dataset):
                 * flux_ul : upper limit map.
 
         """
-        if dataset.stat_type != "cash":
-            raise TypeError(f"{type(dataset)} is not a valid type for {self.__class__}")
-        dataset_models = dataset.models
+        if not isinstance(datasets, (Datasets, list)):
+            datasets = Datasets([datasets])
+
+        geom_ref = datasets[0].counts.geom
+        for dataset in datasets:
+            if dataset.stat_type != "cash":
+                raise TypeError(
+                    f"{type(dataset)} is not a valid type for {self.__class__}"
+                )
+            if dataset.counts.geom != geom_ref:
+                raise TypeError("Datasets geometries must match")
 
-        pad_width = self.estimate_pad_width(dataset=dataset)
-        dataset = dataset.pad(pad_width, name=dataset.name)
-        dataset = dataset.downsample(self.downsampling_factor, name=dataset.name)
+        datasets_models = datasets.models
 
-        energy_axis = self._get_energy_axis(dataset=dataset)
+        pad_width = (0, 0)
+        for dataset in datasets:
+            pad_width_dataset = self.estimate_pad_width(dataset=dataset)
+            pad_width = tuple(np.maximum(pad_width, pad_width_dataset))
+
+        datasets_padded = Datasets()
+        for dataset in datasets:
+            dataset = dataset.pad(pad_width, name=dataset.name)
+            dataset = dataset.downsample(self.downsampling_factor, name=dataset.name)
+            datasets_padded.append(dataset)
+        datasets = datasets_padded
```

Comment:
you could do this in one for loop instead of two

Reply:
I think he uses the first loop to determine the max. width for the padding across all datasets first. Then the same pad width is used later for all datasets. I think the two loop approach is correct.

Reply:
ah ok thank you now I see